packages/rich-text-links/src/index.ts

```diff
@@ -30,7 +31,7 @@ export function getRichTextResourceLinks(
   nodeType: AcceptedResourceLinkTypes,
 ): ResourceLink[] {
   const links = new Map<string, ResourceLink>();
-  const isValidType = (actualNodeType: BLOCKS, data: NodeData) =>
+  const isValidType = (actualNodeType: string, data: NodeData) =>
```

Comment:
Since the `node` can have a different type than `Block` the actualNodeType does not satisfy `BLOCKS` but it would be `BLOCKS | INLINES | 'Text'`, which makes it easier to type as `string` as the narrow type has not much value here

Reply:
I'd argue it clears the intention what node types can be, `string` is very generic.

Reply:
It's the same type as on the interface: 
```
export interface Node {
    readonly nodeType: string;
    data: NodeData;
}
```
Before this we incorrectly type-casted the `Inline | Block | Text` node to `Block` and used the type from the `Block` interface
```
export interface Block extends Node {
    nodeType: BLOCKS;
    content: Array<Block | Inline | Text>;
}
```